xl-v1-multipart.go

```diff
@@ -631,9 +631,11 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 
 	// Rename if an object already exists to temporary location.
 	uniqueID := getUUID()
-	err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
-	if err != nil {
-		return "", toObjectErr(err, bucket, object)
+	if xl.isObject(bucket, object) {
+		err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
```

Comment:
> why would there be a rename on a/b/ ? when the object is a/b/c/d, seems like the bug is somewhere else.

check the unit test case.
1) create object a/b/c/d
2) create object a/b - now this is expected fail - because ultimately when we do renam(tmp, "a/b') the rename would fail complaining that distination directory "a/b" already exists.

But create object "a/b" is succeeding because we would have renamed original "a/b" to a tmp path. (we should rename only if it was an object, not prefix of an object)